components/ledger/ledger-utxo-token-cache/src/main/kotlin/net/corda/ledger/utxo/token/cache/services/PerformanceClaimStateStoreImpl.kt

```diff
@@ -0,0 +1,113 @@
+package net.corda.ledger.utxo.token.cache.services
+
+import net.corda.data.ledger.utxo.token.selection.state.TokenPoolCacheState
+import net.corda.ledger.utxo.token.cache.entities.TokenPoolKey
+import net.corda.libs.statemanager.api.State
+import net.corda.libs.statemanager.api.StateManager
+import net.corda.utilities.time.Clock
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.RejectedExecutionHandler
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+
+@Suppress("UNUSED")
+class PerformanceClaimStateStoreImpl(
+    private val key: TokenPoolKey,
+    storedPoolClaimState: StoredPoolClaimState,
+    private val serialization: TokenPoolCacheStateSerialization,
+    private val stateManager: StateManager,
+    private val clock: Clock
+) : ClaimStateStore {
+
+    private var currentState = storedPoolClaimState
+    private var executor = ThreadPoolExecutor(
+        1, 1,
+        0L, TimeUnit.MILLISECONDS,
+        LinkedBlockingQueue(1),
+        ThreadPoolExecutor.DiscardPolicy()
```

Comment:
Probably needs a comment here why discarding is safe, e.g.:
```suggestion
        // Since this pool is used to drain the queue and each attempt to do so carries on till the queue is empty, it is OK to discard an invocation if there is an intention to perform draining attempt already.
        ThreadPoolExecutor.DiscardPolicy()
```